### PR TITLE
fix: handle DV mask exhaustion and short masks in batch_project

### PR DESCRIPTION
**Pull request by ethan-tyler**
# Description
This fixes deletion vector selection mask handling across RecordBatch boundaries during scans.

- Add DvMaskResult and consume_dv_mask() to encapsulate mask consumption logic
-  Remove exhausted selection vector entries from DashMap after use
- Pad short masks with true (keep rows) instead of erroring
- Guard against empty batches to prevent panic in extract_file_id
- Add unit tests for mask drain, exhaustion, and empty batch handling

# Related Issue(s)
- Relates to #1094

# Documentation

### CODE REVIEW

File: crates/core/src/delta_datafusion/table_provider/next/scan/exec.rs [hunk 1]
```diff
@@ -329,25 +363,29 @@ impl DeltaScanStream {
...
         {
-            if selection_vector.len() >= batch.num_rows() {
-                let sv: Vec<bool> = selection_vector.drain(0..batch.num_rows()).collect();
-                Some(sv)
-            } else {
-                let remaining = batch.num_rows() - selection_vector.len();
-                let sel_len = selection_vector.len();
-                let mut sv: Vec<bool> = selection_vector.drain(0..sel_len).collect();
-                sv.extend(vec![true; remaining]);
-                Some(sv)
```
Comment (roeap) [inline crates/core/src/delta_datafusion/table_provider/next/scan/exec.rs:344]: just did a quick pass. would it already suffice to remove the vector from the map once we needed to extend it?

Comment (ethan-tyler) [reply crates/core/src/delta_datafusion/table_provider/next/scan/exec.rs:344]: Thanks for the response and quick pass! Removing on the extend path helps, but it doesn’t cover the case where we drain exactly batch.num_rows() and the vector becomes empty, or when an empty vector is already present. The fix needs to remove entries whenever the selection vector is exhausted, regardless of which branch got us there.

Comment (roeap) [reply crates/core/src/delta_datafusion/table_provider/next/scan/exec.rs:344]: makes sense! approving!
